index.js

```diff
@@ -121,6 +121,7 @@ function serialize(name, val, options) {
   if (null != opt.maxAge) {
     var maxAge = opt.maxAge - 0;
     if (isNaN(maxAge)) throw new Error('maxAge should be a Number');
+    if (maxAge === Infinity || maxAge === -Infinity) throw new Error("Infinity is invalid value for maxAge")
```

Comment:
```suggestion
    if (!Number.isFinite(maxAge)) throw new Error('maxAge must be finite')
```

Reply:
I first thought of this as well, but MDN suggested that `Number.isFinite` isn't supported in IE.

I know this package is meant to be used serverside, and also I'm not sure at all if there are other things in here that IE doesn't support, but I figured there's likely someone running this in the frontend somewhere.

Reply:
That seems like a fair thought, but (1) we don't support web browsers unless they are part of the module's CI and (2) this module seems almost useless in the browser context, as it is about reading cookies browsers _send_ and writing cookies browsers _receive_; it has no functionality for writing cookies browsers need to send or reading cookies browsers would receive.

Reply:
Perhaps just use the `isFinite` global function?

Reply:
```suggestion
    if (!isFinite(maxAge)) throw new Error('maxAge must be finite')
```

Reply:
That sounds like a good compromise 👍 

Here's at least one [package](https://www.npmjs.com/package/universal-cookie) that seems to be using this in the Browser 😨

Reply:
Ultimately, this package is under the semver 0.y.z rules right now.

Reply:
They're using it to [parse](https://github.com/reactivestack/cookies/blob/bb68224696a59d8089fa85aa4f116c205ff31a92/packages/universal-cookie/src/utils.ts#L22) and [serialize](https://github.com/reactivestack/cookies/blob/master/packages/universal-cookie/src/Cookies.ts#L79) cookies, apparently

Reply:
Since this module is under the semver 0.y.z rules, there are no support rules; if they don't want to see it break, the best option here is that could contribute a web browser test suite to the module, otherwise we cannot properly make changes to this module without possibly breaking them. For all we know, they don't even support IE themselves, meaning if we broke IE, they wouldn't care.

It's one thing to make decisions in our module within the constraints we set on ourselves (Node.js versions supported, adherence to SemVer, etc.) but we can't really be upheld to unknown constraints (like whatever web browser environments are out there) since we don't know what folks even need on that front.